base_layer/core/src/transactions/transaction_components/output_flags.rs

```diff
@@ -31,15 +31,20 @@ use crate::consensus::{ConsensusDecoding, ConsensusEncoding, ConsensusEncodingSi
 
 bitflags! {
     #[derive(Deserialize, Serialize)]
-    pub struct OutputFlags: u8 {
-        /// Output is a coinbase output, must not be spent until maturity
-        const COINBASE_OUTPUT      = 0b0000_0001;
-        const NON_FUNGIBLE         = 0b0000_1000;
-        const ASSET_REGISTRATION   = 0b0000_0010 | Self::NON_FUNGIBLE.bits;
-        const MINT_NON_FUNGIBLE    = 0b0000_0100 | Self::NON_FUNGIBLE.bits;
-        const BURN_NON_FUNGIBLE    = 0b1000_0000 | Self::NON_FUNGIBLE.bits;
-        const SIDECHAIN_CHECKPOINT = 0b0001_0000 | Self::NON_FUNGIBLE.bits;
-        const COMMITTEE_DEFINITION = 0b0010_0000 | Self::NON_FUNGIBLE.bits;
+    pub struct OutputFlags: u16 {
+        /// Output is a coinbase output, must not be spent until maturity.
+        const COINBASE_OUTPUT       = 0x0001;
+        /// Output defines a side-chain contract.
+        const CONTRACT_DEFINITION   = 0x0100;
+        /// Output defines the constitution for a side-chain contract.
+        const CONTRACT_CONSTITUTION = 0x0200;
+        /// Output signals validator node acceptance to run a contract.
+        const CONTRACT_ACCEPT       = 0x0400;
+        /// Output is a contract checkpoint.
+        const CONTRACT_CHECKPOINT   = 0x0800;
```

Comment:
Isn't that going to be the first checkpoint? Or does it need to be a custom output for that?

Reply:
I think you're right and we might need an abandoned flag so the asset owner can spend an abandoned checkpoint into that state to activate the emergency keys

Reply:
Agreed

Reply:
I left off the INITIAL_CHECKPOINT flag for now because even if we have it, we'd have to check that it is an INITIAL checkpoint (we can't trust the flag), in doing so we have done the work to detect it.